### PR TITLE
Fix build errors and a crash in xcode 15

### PR DESCRIPTION
**Pull request by cacmartinez**
Fixes build errors when trying to compile with xcode15. 

It also fixes a crash happening on XCode15 when attempting to create a new image with either a height or width of 0 or less (UIGraphicsBeginImageContextWithOptions). This can happen when isCancelled returns true.

### CODE REVIEW

File: Source/Details/ASGraphicsContext.mm [hunk 1]
```diff
@@ -121,7 +121,9 @@ NS_INLINE void ASConfigureExtendedRange(UIGraphicsImageRendererFormat *format)
   }
 
   // Bad OS or experiment flag. Use UIGraphics* API.
-  UIGraphicsBeginImageContextWithOptions(size, opaque, scale);
+  if (size.width > 0 && size.height > 0) {
+    UIGraphicsBeginImageContextWithOptions(size, opaque, scale);
```
Comment (rcancro) [inline Source/Details/ASGraphicsContext.mm:125]: If we have an invalid size, I think we should exit early and return nil. Shortly after starting the image context we call `UIGraphicsGetImageFromCurrentImageContext` and `UIGraphicsEndImageContext`. I'm not sure what the behavior is if we end a graphics context without starting one.

Comment (cacmartinez) [reply Source/Details/ASGraphicsContext.mm]: I've switched to an early return

**Review by rcancro — APPROVED** [review]

**Comment by rahulVermaSimpplr** [comment]
Is this code is also available in the released version that is 3.1.0

**Comment by keerthika-priya** [comment]
Is this issue happening only if the app is built using XCode 15?

**Pull request merged**